static/app/utils/api/useFetchSequentialPages.spec.tsx

```diff
@@ -0,0 +1,228 @@
+    expect(getQueryKey).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call the queryFn 1 times when the first response has no next page', async () => {
+    MockApiClient.addMockResponse({
+      url: MOCK_API_ENDPOINT,
+      body: 'text result',
+      headers: {Link: getPaginationPageLink({numRows: 0, pageSize: 10, offset: 0})},
+    });
+    const getQueryKey = queryKeyFactory();
+
+    const {result, waitFor} = reactHooks.renderHook(useFetchSequentialPages, {
+      wrapper: makeWrapper(makeTestQueryClient()),
+      initialProps: {
+        enabled: true,
+        getQueryKey,
+        perPage: 10,
+      },
+    });
+
+    expect(result.current.isFetching).toBeTruthy();
+    await waitFor(() => expect(result.current.isFetching).toBeFalsy());
+    expect(getQueryKey).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call the queryFn N times, until the response has no next page', async () => {
+    MockApiClient.addMockResponse({
+      url: MOCK_API_ENDPOINT,
+      body: 'results starting at 0',
+      match: [MockApiClient.matchQuery({cursor: '0:0:0', per_page: 10})],
+      headers: {Link: getPaginationPageLink({numRows: 13, pageSize: 10, offset: 0})},
+    });
+    MockApiClient.addMockResponse({
+      url: MOCK_API_ENDPOINT,
+      body: 'results starting at 10',
+      match: [MockApiClient.matchQuery({cursor: '0:10:0', per_page: 10})],
+      headers: {Link: getPaginationPageLink({numRows: 13, pageSize: 10, offset: 10})},
+    });
+    const getQueryKey = queryKeyFactory();
+
+    const {result, waitFor} = reactHooks.renderHook(useFetchSequentialPages, {
+      wrapper: makeWrapper(makeTestQueryClient()),
+      initialProps: {
+        enabled: true,
+        getQueryKey,
+        perPage: 10,
+      },
+    });
+
+    expect(result.current.isFetching).toBeTruthy();
+    await waitFor(() => expect(result.current.isFetching).toBeFalsy());
+    expect(getQueryKey).toHaveBeenCalledTimes(2);
+  });
+
+  it('should return a list of all pages that have been resolved', async () => {
+    MockApiClient.addMockResponse({
+      url: MOCK_API_ENDPOINT,
+      body: 'results starting at 0',
+      match: [MockApiClient.matchQuery({cursor: '0:0:0', per_page: 10})],
+      headers: {Link: getPaginationPageLink({numRows: 13, pageSize: 10, offset: 0})},
+    });
+    MockApiClient.addMockResponse({
+      url: MOCK_API_ENDPOINT,
+      body: 'results starting at 10',
+      match: [MockApiClient.matchQuery({cursor: '0:10:0', per_page: 10})],
+      headers: {Link: getPaginationPageLink({numRows: 13, pageSize: 10, offset: 10})},
+    });
+    const getQueryKey = queryKeyFactory();
+
+    const {result, waitFor} = reactHooks.renderHook(useFetchSequentialPages, {
+      wrapper: makeWrapper(makeTestQueryClient()),
+      initialProps: {
+        enabled: true,
+        getQueryKey,
+        hits: 13,
+        perPage: 10,
+      },
+    });
+
+    await waitFor(() => expect(result.current.isFetching).toBeFalsy());
```

Comment:
nice waitFor instead of waitForNextUpdate

Reply:
i'm learning!